src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/roll/roll_kernel_ref.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "roll_kernel_ref.h"
+
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+namespace {
+
+CommonDispatchData SetDefault(const roll_params& params) {
+    CommonDispatchData dispatch_data;
+    const auto in_layout = params.inputs[0].GetLayout();
+    const auto out_layout = params.outputs[0].GetLayout();
+    std::vector<std::vector<Tensor::DataChannelName>> dims_by_gws;
+
+    const auto& output = params.outputs[0];
+
+    switch (out_layout) {
+    case DataLayout::bfyx:
+        dispatch_data.gws = {output.X().v, output.Y().v, output.Feature().v * output.Batch().v};
+        dims_by_gws = {{Tensor::DataChannelName::X},
+                       {Tensor::DataChannelName::Y},
+                       {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
+        break;
+    case DataLayout::bfzyx:
+        dispatch_data.gws = {output.X().v * output.Y().v, output.Z().v, output.Feature().v * output.Batch().v};
+        dims_by_gws = {{Tensor::DataChannelName::X, Tensor::DataChannelName::Y},
+                       {Tensor::DataChannelName::Z},
+                       {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
+        break;
+    case DataLayout::bfwzyx:
+        dispatch_data.gws = {output.X().v * output.Y().v,
+                             output.Z().v * output.W().v,
+                             output.Feature().v * output.Batch().v};
+        dims_by_gws = {{Tensor::DataChannelName::X, Tensor::DataChannelName::Y},
+                       {Tensor::DataChannelName::Z, Tensor::DataChannelName::W},
+                       {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
+        break;
+    default:
```

Comment:
When you add support for blocked formats, please add hadndle of those formats here too

Reply:
(As well as the white list of "enable_fsv16" in program.cpp too)

Reply:
agreed not to do now